#### The World Bank's international debt data

<p>It's not that we humans only take debts to manage our necessities. A country may also take debt to manage its economy. For example, infrastructure spending is one costly ingredient required for a country's citizens to lead comfortable lives. <a href="https://www.worldbank.org">The World Bank</a> is the organization that provides debt to countries.</p>

<p>In this notebook, I am going to analyze international debt data collected by The World Bank. The dataset contains information about the amount of debt (in USD) owed by developing countries across several categories. I will be finding the answers to questions like: </p>

<ul>
<li>What is the total amount of debt that is owed by the countries listed in the dataset?</li>
<li>Which country owns the maximum amount of debt and what does that amount look like?</li>
<li>What is the average amount of debt owed by countries across different debt indicators?</li>
</ul>


In [1]:
# load the SQL extension for running SQL queries in the notebook
%load_ext sql

# %reload_ext sql

In [22]:
%%sql
postgresql://postgres:****@localhost:5432/postgres 

'Connected: postgres@postgres'

In [23]:
%%sql 
SELECT * FROM international_debt LIMIT 10


 * postgresql://postgres:***@localhost:5432/postgres
10 rows affected.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('72894453.700000003')),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('53239440.100000001')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('61739336.899999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('49114729.399999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('39903620.100000001')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', Decimal('39107845')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', Decimal('23779724.300000001')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', Decimal('13335820')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', Decimal('100847181.900000006')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', Decimal('72894453.700000003'))]

In [24]:
%%sql 
SELECT 
    column_name, data_type
FROM
    information_schema.columns
WHERE
    table_name = 'international_debt'

 * postgresql://postgres:***@localhost:5432/postgres
5 rows affected.


[('debt', 'numeric'),
 ('country_name', 'character varying'),
 ('country_code', 'character varying'),
 ('indicator_name', 'text'),
 ('indicator_code', 'text')]

#### Finding the number of distinct countries
<p>From the first ten rows, I can see the amount of debt owed by Afghanistan in the different debt indicators. However, I do not know the number of different countries that are present in the table. There are repetitions in the country names because a country is most likely to have debt in more than one debt indicator.</p>

<p>Without a count of unique countries, I will not be able to perform my statistical analyses holistically. In this section, I am going to extract the number of unique countries present in the table."</p>




In [25]:
%%sql
SELECT COUNT(DISTINCT country_name) as total_distinct_countries 
FROM international_debt

 * postgresql://postgres:***@localhost:5432/postgres
1 rows affected.


[(124,)]

In [28]:
%%sql
SELECT DISTINCT country_name 
FROM international_debt
LIMIT 5

 * postgresql://postgres:***@localhost:5432/postgres
5 rows affected.


[('Indonesia',),
 ('Bangladesh',),
 ('Iran, Islamic Rep.',),
 ('Cameroon',),
 ('St. Lucia',)]

#### Finding out the distinct debt indicators

<p>I can see that there are a total of 124 countries present on the table. As I saw in the first section, there is a column called <code>indicator_name</code> that briefly specifies the purpose of taking the debt. Just beside that column, there is another column called <code>indicator_code</code> which symbolizes the category of these debts. Knowing about these various debt indicators will help me to understand the areas in which a country can possibly be indebted to.</p>

In [29]:
%%sql
SELECT DISTINCT(indicator_code) as distinct_debt_indicators,indicator_name
FROM international_debt
ORDER BY distinct_debt_indicators

 * postgresql://postgres:***@localhost:5432/postgres
25 rows affected.


[('DT.AMT.BLAT.CD', 'PPG, bilateral (AMT, current US$)'),
 ('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)'),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)'),
 ('DT.AMT.MLAT.CD', 'PPG, multilateral (AMT, current US$)'),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)'),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)'),
 ('DT.AMT.PCBK.CD', 'PPG, commercial banks (AMT, current US$)'),
 ('DT.AMT.PROP.CD', 'PPG, other private creditors (AMT, current US$)'),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)'),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)'),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)'),
 ('DT.DIS.MLAT.CD', 'PPG, multilateral (DIS, current US$)'),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)'),
 ('DT.DIS.PCBK.CD', 'PPG, commercial banks (DIS, current US$)'),
 ('DT.DIS.PROP.CD', 'PPG, other private creditors (DIS, current US$)'),
 ('DT.DIS.PRVT.CD', 'PPG, private creditors (DIS, current US$)'),
 ('DT.INT.BLAT.CD', 'PPG, bilateral (INT, current US$)'),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)'),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)'),
 ('DT.INT.MLAT.CD', 'PPG, multilateral (INT, current US$)'),
 ('DT.INT.OFFT.CD', 'PPG, official creditors (INT, current US$)'),
 ('DT.INT.PBND.CD', 'PPG, bonds (INT, current US$)'),
 ('DT.INT.PCBK.CD', 'PPG, commercial banks (INT, current US$)'),
 ('DT.INT.PROP.CD', 'PPG, other private creditors (INT, current US$)'),
 ('DT.INT.PRVT.CD', 'PPG, private creditors (INT, current US$)')]

#### Totaling the amount of debt owed by the countries

<p>As mentioned earlier, the financial debt of a particular country represents its economic state. However, if I were to project this on an overall global scale, how would I approach it?</p>

<p>Let's switch gears from the debt indicators now and find out the total amount of debt (in USD) that is owed by the different countries. This will give me a sense of how the overall economy of the entire world is holding up.</p>

In [30]:
%%sql
SELECT ROUND(SUM(debt)/1000000,2) AS total_debt
FROM international_debt

 * postgresql://postgres:***@localhost:5432/postgres
1 rows affected.


[(Decimal('3079734.49'),)]

#### Country with the highest debt

<p>Now that I have the exact total of the amounts of debt owed by several countries, let's find out the country that owns the highest amount of debt along with the amount. It's important to note that this debt is the sum of different debts owed by a country across several categories. This will help me to understand more about the country in terms of its socio-economic scenarios. I can also find out the category in which the country owns its highest debt, but I will leave that for now.</p>

In [32]:
%%sql

SELECT country_name, SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC
LIMIT 5;


 * postgresql://postgres:***@localhost:5432/postgres
5 rows affected.


[('China', Decimal('285793494734.200001568')),
 ('Brazil', Decimal('280623966140.800007581')),
 ('South Asia', Decimal('247608723990.600003211')),
 ('Least developed countries: UN classification', Decimal('212880992791.900000988')),
 ('Russian Federation', Decimal('191289057259.200001943'))]

#### Average amount of debt across indicators
<p>So, it was <em>China</em>. A more in-depth breakdown of China's debts can be found <a href="https://datatopics.worldbank.org/debt/ids/country/CHN">here</a>. </p>
<p>I now have a brief overview of the dataset and a few of its summary statistics. I already have an idea of the different debt indicators in which the countries owe their debts. I can dig even further to find out on an average how much debt a country owes? This will give me a better sense of the distribution of the amount of debt across different indicators.</p>

In [34]:
%%sql
SELECT 
    indicator_code AS debt_indicator,
    indicator_name,
    AVG(debt) AS average_debt
FROM international_debt
GROUP BY debt_indicator, indicator_name
ORDER BY average_debt DESC
LIMIT 10;



 * postgresql://postgres:***@localhost:5432/postgres
10 rows affected.


[('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('5904868401.499193612')),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', Decimal('5161194333.812658349')),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)', Decimal('2152041216.890243888')),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('1958983452.859836046')),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', Decimal('1803694101.963265321')),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)', Decimal('1644024067.650806481')),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)', Decimal('1223139290.398230108')),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', Decimal('1220410844.421518983')),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', Decimal('1191187963.083064523')),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)', Decimal('1082623947.653623188'))]

#### The highest amount of principal repayments
<p>The indicator <code>DT.AMT.DLXF.CD</code> tops the chart of average debt. This category includes repayment of long term debts. Countries take on long-term debt to acquire immediate capital. More information about this category can be found <a href="https://datacatalog.worldbank.org/principal-repayments-external-debt-long-term-amt-current-us-0">here</a>. </p>

<p>An interesting observation in the above finding is that there is a huge difference in the amounts of the indicators after the second one. This indicates that the first two indicators might be the most severe categories in which the countries owe their debts.</p>

<p>I can investigate this a bit more so as to find out which country owes the highest amount of debt in the category of long term debts (<code>DT.AMT.DLXF.CD</code>). Since not all the countries suffer from the same kind of economic disturbances, this finding will allow me to understand that particular country's economic condition a bit more specifically. </p>

In [40]:
%%sql
SELECT 
    country_name, 
    indicator_name,
    debt as maximum_debt
FROM international_debt
WHERE debt = (SELECT 
                  MAX(debt)
              FROM international_debt
              WHERE indicator_code='DT.AMT.DLXF.CD');

 * postgresql://postgres:***@localhost:5432/postgres
1 rows affected.


[('China', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('96218620835.699996948'))]

#### The most common debt indicator
<p>China has the highest amount of debt in the long-term debt (<code>DT.AMT.DLXF.CD</code>) category. This is verified by <a href="https://data.worldbank.org/indicator/DT.AMT.DLXF.CD?end=2018&most_recent_value_desc=true">The World Bank</a>.

<p>I saw that long-term debt is the topmost category when it comes to the average amount of debt. But is it the most common indicator in which the countries owe their debt? Let's find that out. </p>

In [36]:
%%sql
SELECT 
    indicator_code,
    COUNT(indicator_code) AS indicator_count
FROM international_debt
GROUP BY indicator_code
ORDER BY indicator_count DESC, indicator_code DESC
LIMIT 20;

 * postgresql://postgres:***@localhost:5432/postgres
20 rows affected.


[('DT.INT.OFFT.CD', 124),
 ('DT.INT.MLAT.CD', 124),
 ('DT.INT.DLXF.CD', 124),
 ('DT.AMT.OFFT.CD', 124),
 ('DT.AMT.MLAT.CD', 124),
 ('DT.AMT.DLXF.CD', 124),
 ('DT.DIS.DLXF.CD', 123),
 ('DT.INT.BLAT.CD', 122),
 ('DT.DIS.OFFT.CD', 122),
 ('DT.AMT.BLAT.CD', 122),
 ('DT.DIS.MLAT.CD', 120),
 ('DT.DIS.BLAT.CD', 113),
 ('DT.INT.PRVT.CD', 98),
 ('DT.AMT.PRVT.CD', 98),
 ('DT.INT.PCBK.CD', 84),
 ('DT.AMT.PCBK.CD', 84),
 ('DT.INT.DPNG.CD', 79),
 ('DT.AMT.DPNG.CD', 79),
 ('DT.INT.PBND.CD', 69),
 ('DT.AMT.PBND.CD', 69)]

#### Other viable debt issues and conclusion
<p>There are a total of six debt indicators in which all the countries listed in our dataset have taken debt. The indicator <code>DT.AMT.DLXF.CD</code> is also there in the list. So, this gives us a clue that all these countries are suffering from a common economic issue. But that is not the end of the story, but just a part of the story.</p>

<p>I'm going to change tracks from <code>debt_indicator</code> now and focus on the amount of debt again - focusing on the maximum amount of debt that each country has. With this, I'll be in a position to identify the other plausible economic issues a country might be going through.</p>

<p>In this notebook, I took a look at debt owed by countries across the globe. Extracted a few summary statistics from the data and unraveled some interesting facts and figures. Also validated the findings to make sure the investigations are correct.</p>

In [45]:
%%sql

SELECT country_name, ROUND(MAX(debt),2) as maximum_debt
FROM international_debt
GROUP BY country_name
ORDER BY maximum_debt DESC
LIMIT 10;

 * postgresql://postgres:***@localhost:5432/postgres
10 rows affected.


[('China', Decimal('96218620835.70')),
 ('Brazil', Decimal('90041840304.10')),
 ('Russian Federation', Decimal('66589761833.50')),
 ('Turkey', Decimal('51555031005.80')),
 ('South Asia', Decimal('48756295898.20')),
 ('Least developed countries: UN classification', Decimal('40160766261.60')),
 ('IDA only', Decimal('34531188113.20')),
 ('India', Decimal('31923507000.80')),
 ('Indonesia', Decimal('30916112653.80')),
 ('Kazakhstan', Decimal('27482093686.40'))]